### PR TITLE
Drop SendGrid email transport support

### LINKED ISSUE #23217: Consider dropping the dependency on nodemailer-sendgrid
### Describe the Bug

directus-api depends on https://github.com/nodemailer/nodemailer-sendgrid which is no longer maintained example https://github.com/nodemailer/nodemailer-sendgrid/pull/15

I'm not aware of a way to exploit this but Directus come up as a source of vulnerabilities in static analysis from tools like Snyk and Wiz.

This package depends on old and deprecated code like https://www.npmjs.com/package/request and vulnerable code as well like https://github.com/advisories/GHSA-72xf-g2v4-qvf3

Sendgrid's doc point to a different solution https://www.twilio.com/en-us/blog/send-smtp-emails-node-js-sendgrid that doesn't require nodemailer-sendgrid and from my unexperienced point of view doesn't affect the way how directus work.

### To Reproduce

npm audit

### Directus Version

all

### Hosting Strategy

Self-Hosted (Custom)

### Database

_No response_

### PR DESCRIPTION
**Pull request by florian-strasser**
Based on issue [#23217](https://github.com/directus/directus/issues/23217) I removed the nodemailer-sendgrid else if branch

What's changed:

- We are dropping support sendgrid

## Potential Risks / Drawbacks

- It's a breaking change that forces users that rely on this to switch to SMTP

## Review Notes / Questions

- nodemailer/sendgrid received the last update 6 years ago which should make clear that this absolutely outdated: [https://github.com/nodemailer/nodemailer-sendgrid](https://github.com/nodemailer/nodemailer-sendgrid)

---

Closes #23217

### CODE REVIEW

**Comment by McSundae** [comment]
I would also remove the package `api/package.json`, but I'm not sure if it is currently used in any other file than `api/src/mailer.ts`

**Comment by McSundae** [comment]
Not sure how to fix the current errors, probably I should not just remove the line from package.json.

 ` WARN  There are cyclic workspace dependencies: /home/runner/work/directus/directus/api, /home/runner/work/directus/directus/directus` 

`ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up to date with api/package.json`

**Comment by paescuj** [comment]
> Not sure how to fix the current errors, probably I should not just remove the line from package.json.
> 
>  ` WARN  There are cyclic workspace dependencies: /home/runner/work/directus/directus/api, /home/runner/work/directus/directus/directus` 
> 
> `ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up to date with api/package.json`

Heya @McSundae 👋 You need to run `pnpm install` to update the lockfile after the change to the `package.json` file.

File: .changeset/hot-actors-lay.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@directus/api": patch
+---
+
+Drop nodemailer-sendgrid support since the package is no longer maintained
```
Comment (paescuj) [inline .changeset/hot-actors-lay.md:5]: This is a major change to the `@directus/api` package

```suggestion
---
"@directus/api": major
---

Dropped support for the SendGrid email transport option
```

**Review by paescuj — CHANGES_REQUESTED** [review]
We also need to mention this on the "Breaking Changes" page [^1][^2], advising users to switch to SMTP as per https://github.com/directus/directus/issues/23217#issuecomment-2271989983.

[^1]: https://docs.directus.io/releases/breaking-changes.html
[^2]: https://github.com/directus/directus/blob/1a2a085731638bf9dff87267b3efee02d7533c2a/docs/releases/breaking-changes.md

File: .changeset/hot-actors-lay.md (hunk 1 above)
Comment (McSundae) [reply .changeset/hot-actors-lay.md]: As suggested I made the changes

**Review by rijkvanzanten — APPROVED** [review]

**Comment by u12206050** [comment]
WHY?!? More importantly why not release this on a minor release, it is a big breaking change to anyone using Sendgrid.

**Comment by McSundae** [comment]
@u12206050 I unterstand your frustration, but as mentioned in the initial issue the package that was used for this is not maintained since over 2 years, so it is a security issue to keep it inside of directus. We recommend to use smtp to connect with send grid

**Comment by u12206050** [comment]
> @u12206050 I understand your frustration, but as mentioned in the initial issue the package that was used for this is not maintained since over 6 years, so it is a security issue to keep it inside of directus. We recommend to use smtp to connect with send grid.
> 
> The package I'm talking about is: https://github.com/nodemailer/nodemailer-sendgrid

Thanks, that I understand, although I still don't understand how this could be released as a patch release, are we too assume from now on that patch releases could also contain breaking changes?

**Comment by jogi-k** [comment]
Same opinion. Why are you bringing out breaking changes in hotfixes?   
Thats completely incompatible with [semantic versioning](https://semver.org/)

**Comment by inovosel** [comment]
Have you added any guide/instructions how to get SendGrid working again after you have removed support for SendGrid transport? Or a way how to manually add SendGrid transport support (with me taking all the risks for using outdated component)?

**Comment by McSundae** [comment]
As already mentioned it is recommended to use simply smtp. So you simply go into your config file and change the specific parameters:

https://docs.directus.io/self-hosted/config-options.html#smtp-smtp

**Pull request merged**